src/app/_api/fetchGlobals.ts

```diff
@@ -1,108 +1,108 @@
-      });
+        if (res?.errors) throw new Error(res?.errors[0]?.message || 'Error fetching settings')
+        return res.data?.Settings
+      })
   } catch (err: unknown) {
-    throw new Error("Error fetching settings")
+    throw new Error('Error fetching settings')
   }
 }
 
 export async function fetchHeader(): Promise<Header> {
-  if (!GRAPHQL_API_URL) throw new Error("NEXT_PUBLIC_SERVER_URL not found");
+  if (!GRAPHQL_API_URL) throw new Error('NEXT_PUBLIC_SERVER_URL not found')
 
   try {
     return await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
-      method: "POST",
+      method: 'POST',
       headers: {
-        "Content-Type": "application/json",
+        'Content-Type': 'application/json',
       },
-      cache: "no-store",
+      cache: 'no-store',
       body: JSON.stringify({
         query: HEADER_QUERY,
       }),
     })
       ?.then(res => {
-        if (!res.ok) throw new Error("Error fetching doc");
-        return res.json();
+        if (!res.ok) throw new Error('Error fetching doc')
+        return res.json()
       })
       ?.then(res => {
-        if (res?.errors) throw new Error(res?.errors[0]?.message || "Error fetching header");
-        return res.data?.Header;
-      });
+        if (res?.errors) throw new Error(res?.errors[0]?.message || 'Error fetching header')
+        return res.data?.Header
+      })
   } catch (err: unknown) {
-    throw new Error("Error fetching header");
+    throw new Error('Error fetching header')
   }
 }
 
 export async function fetchFooter(): Promise<Footer> {
-  if (!GRAPHQL_API_URL) throw new Error("NEXT_PUBLIC_SERVER_URL not found");
+  if (!GRAPHQL_API_URL) throw new Error('NEXT_PUBLIC_SERVER_URL not found')
 
   try {
     return await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
-      method: "POST",
+      method: 'POST',
       headers: {
-        "Content-Type": "application/json",
+        'Content-Type': 'application/json',
       },
       body: JSON.stringify({
         query: FOOTER_QUERY,
       }),
     })
       .then(res => {
-        if (!res.ok) throw new Error("Error fetching doc");
-        return res.json();
+        if (!res.ok) throw new Error('Error fetching doc')
+        return res.json()
       })
       ?.then(res => {
-        if (res?.errors) throw new Error(res?.errors[0]?.message || "Error fetching footer");
-        return res.data?.Footer;
-      });
+        if (res?.errors) throw new Error(res?.errors[0]?.message || 'Error fetching footer')
+        return res.data?.Footer
+      })
   } catch (err: unknown) {
-    throw new Error("Error fetching footer");
+    throw new Error('Error fetching footer')
   }
 }
 
 export async function fetchSocials(): Promise<Social> {
-  if (!GRAPHQL_API_URL) throw new Error("NEXT_PUBLIC_SERVER_URL not found");
+  if (!GRAPHQL_API_URL) throw new Error('NEXT_PUBLIC_SERVER_URL not found')
 
   try {
     return await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
-      method: "POST",
+      method: 'POST',
```

Comment:
This is pretty repetitive, it would be good to use a helper function for this (eventually)

Reply:
yep, `fetcher` (the one used to pull the content grid) will eventually be incorporated in all `fetchXXX` functions
planned for the refactor / next cleanup phase